spec/values/calculation/pow.hrx

```diff
@@ -0,0 +1,192 @@
+<===> README.md
+Most of the same behavior tested for `calc()` applies to `pow()`, but for
+terseness' sake isn't tested explicitly.
+
+<===>
+================================================================================
+<===> positive/input.scss
+a {b: pow(10, 10)}
+
+<===> positive/output.css
+a {
+  b: 10000000000;
+}
+
+<===>
+================================================================================
+<===> base/negative/input.scss
+a {b: pow(-10, 10)}
+
+<===> base/negative/output.css
+a {
+  b: 10000000000;
+}
+
+<===>
+================================================================================
+<===> exponent/negative/input.scss
+a {b: pow(10, -10)}
+
+<===> exponent/negative/output.css
+a {
+  b: 0.0000000001;
+}
+
+<===>
+================================================================================
+<===> zeros/input.scss
+a {b: pow(0, 0)}
+
+<===> zeros/output.css
+a {
+  b: 1;
+}
+
+<===>
+================================================================================
+<===> y_infinity/positive/input.scss
+a {b: pow(infinity, 10)}
+
+<===> y_infinity/positive/output.css
+a {
+  b: calc(infinity);
+}
+
+<===>
+================================================================================
+<===> x_infinity/positive/input.scss
+a {b: pow(10, infinity)}
+
+<===> x_infinity/positive/output.css
+a {
+  b: calc(infinity);
+}
+
+<===>
+================================================================================
+<===> simplification/input.scss
+a {b: pow(3px - 1px + var(--c), 4px + 10px)}
+
+<===> simplification/output.css
+a {
+  b: pow(2px + var(--c), 14px);
+}
+
+<===>
+================================================================================
+<===> error/units/compatible/input.scss
+a {b: pow(10px, 10px)}
+
+<===> error/units/compatible/error
+Error: Expected 10px to have no units.
+  ,
+1 | a {b: pow(10px, 10px)}
+  |       ^^^^^^^^^^^^^^^
+  '
+  input.scss 1:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/units/unknown_and_unitless/input.scss
+a {b: pow(10%, 10)}
+
+<===> error/units/unknown_and_unitless/error
+Error: 10% and 10 are incompatible.
```

Comment:
This error is misleading—the issue is that _no_ units are allowed. Making them compatible wouldn't fix the error.

Reply:
So, I was tracing the error and it seems like the `Expected 10% to have no units.` error is being generated in [this](https://github.com/sass/dart-sass/blob/4d38e3b132dd8b4751c8d1f5d23869593b49a146/lib/src/visitor/async_evaluate.dart#L2385) switch block. The problem is, the block is catching the exception and calling the [`_verifyCompatibleNumbers()`](https://github.com/sass/dart-sass/blob/4d38e3b132dd8b4751c8d1f5d23869593b49a146/lib/src/visitor/async_evaluate.dart#L2413) instead of rethrowing. I can imagine there's a good reason for that so I'm not sure how to proceed. Thoughts? 🤔

Reply:
If you guard that call by checking if the original error message actually mentions the numbers being incompatible, it should avoid changing an unrelated error.

Reply:
Okay, I added an `if` statement checking if the message contains the word `compatible`. So this applies to both compatibility error messages: `1px and $min: 1deg have incompatible units` or `Number 2px*px isn't compatible with CSS calculations.`